src/NodaTime.Test/TimeZones/BclDateTimeZoneTest.cs

```diff
@@ -267,6 +268,50 @@ public void FakeDaylightSavingTime()
             Assert.AreEqual(expectedSummer, summerInterval);
         }
 
+        // See https://github.com/nodatime/nodatime/issues/1524 for the background
+        // It would be nice to test earlier dates (e.g. 1967 and 1998) where transitions
+        // actually occurred on the first of the "next month", but the Windows database has very different
+        // data from TZDB in those cases.
+        [Test]
+        public void TransitionAtMidnight()
+        {
+            var bclZone = GetBclZoneOrIgnore("E. South America Standard Time");
+            var nodaTzdbZone = DateTimeZoneProviders.Tzdb["America/Sao_Paulo"];
+            var nodaBclZone = BclDateTimeZone.FromTimeZoneInfo(bclZone);
+
+            var expectedFallBack = Instant.FromUtc(2012, 2, 26, 2, 0, 0);
+            var expectedSpringForward = Instant.FromUtc(2012, 10, 21, 3, 0, 0);
+
+            // We assume that the fall-back happens before the spring-forward in the dates
+            var betweenTransitions = expectedFallBack + Duration.FromDays(20);
+
+            // Check that these transitions are as expected when we use TZDB.
+            var nodaTzdbInterval = nodaTzdbZone.GetZoneInterval(betweenTransitions);
+            Assert.AreEqual(expectedFallBack, nodaTzdbInterval.Start);
+            Assert.AreEqual(expectedSpringForward, nodaTzdbInterval.End);
+
+            // Check that the real BCL time zone behaves as reported in the issue: the transitions occur one millisecond early
+            var rules = bclZone!.GetAdjustmentRules();
```

Comment:
This isn't really related to this change, but I'm trying to understand why the null-forgiving operator is needed? From what I can see, the only useful use of this would be if we have an explicitly-nullable reference type (`TimeZoneInfo?`), but we don't?

(I'm looking at  https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references#nullable-annotation-context for reference.)

(Likewise a little bit below.)

Reply:
Are - it's not now. It was in an earlier version. The local variable isn't needed either! (Was just useful when debugging)